orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt

```diff
@@ -56,4 +57,22 @@ interface AuthenticationAware {
       ExecutionContext.clear()
     }
   }
+
+  fun backtrackSkippedStages(stage: StageExecution): StageExecution {
+    if (stage.isManualJudgmentType &&
+      !stage.status.isSkipped &&
+      stage.withPropagateAuthentication()) {
+      return stage;
+    }
+    val previousStage = if (stage.ancestors().size > 1) stage.ancestors().get(1) else null
+    previousStage?: return stage
```

Comment:
I could use some comments here explaining the logic.

Reply:
Thanks for that.  I could still use some comments here though.  Or maybe a change to the code would make it easier for me to read/understand without comments.  I think what it's doing is:

* if there's no previous stage, return the current stage
* otherwise, return `backtrackSkippedStages(previousStage);`

For me that would be easier to see with:
```
    if (previousStage == null) {
      return stage;
    }
```
A ternary would be even clearer, at least for me...but since kotlin doesn't have them
```
return if (previousStage == null) stage else backtrackSkippedStages(previousStage);
```

Reply:
I agree, looks more elegant, anyway I added extra comments to clarify for future developers 😃